app/src/main/java/mozilla/lockbox/support/SecurePreferences.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.support
+
+import android.content.SharedPreferences
+import android.util.Base64
+import mozilla.components.lib.dataprotect.Keystore
+import java.nio.charset.StandardCharsets
+
+private const val KEYSTORE_LABEL = "lockbox-keystore"
+private const val BASE_64_FLAGS = Base64.URL_SAFE or Base64.NO_PADDING
+
+open class SecurePreferences(
+    private val keystore: Keystore = Keystore(KEYSTORE_LABEL)
+) {
+    private lateinit var prefs: SharedPreferences
+
+    open fun apply(sharedPreferences: SharedPreferences) {
+        prefs = sharedPreferences
+    }
+
+    open fun getString(key: String): String? {
+        return if (prefs.contains(key)) {
+            val value = prefs.getString(key, "")
+            val encrypted = Base64.decode(value, BASE_64_FLAGS)
+            val plain = keystore.decryptBytes(encrypted)
+
+            String(plain, StandardCharsets.UTF_8)
+        } else {
+            null
+        }
+    }
+
+    open fun putString(key: String, value: String) {
```

Comment:
Nit: try and make this mirror the `SharedPreferences` API as much as possible.

Reply:
This is functionally the same, but I felt like the "editor" concept was pretty high overhead for a two method class.

Reply:
👍 

Eventually, I think we should try and make this a version for `android-components` that is a drop in replacement/wrapper for `SharedPreferences`.